packages/bpk-component-card-list/src/BpkCardList.tsx

```diff
@@ -0,0 +1,165 @@
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ReactElement } from 'react';
+import { useState } from 'react';
+
+import BpkBreakpoint, { BREAKPOINTS } from '../../bpk-component-breakpoint';
+import { BpkButtonV2 } from '../../bpk-component-button';
+import BpkSectionHeader from '../../bpk-component-section-header';
+import { cssModules } from '../../bpk-react-utils';
+
+import BpkCardListGrid from './BpkCardListGrid';
+import BpkCardListRail from './BpkCardListRail';
+import BpkCardListRow from './BpkCardListRow';
+import BpkCardListStack from './BpkCardListStack';
+import { BpkAccessoryTypes, type BpkCardListProps } from './common-types';
+
+import STYLES from './BpkCardList.module.scss';
+
+const getClassName = cssModules(STYLES);
+const MAX_ITEMS = 12; // MAX should be 12 for Desktop Grid and Mobile Stack
+const DEFAULT_ITEMS = 3;
+
+const BpkCardList = (props: BpkCardListProps) => {
+  const {
+    buttonText,
+    cardList,
+    description,
+    initiallyShownCards = DEFAULT_ITEMS,
+    layoutDesktop,
+    layoutMobile,
+    onButtonClick,
+    title,
+  } = props;
+
+  const [visibleCards, setVisibleCards] = useState(
+    cardList.slice(0, initiallyShownCards),
+  );
+  const [collapsed, setCollapsed] = useState(true);
+
+  const showContent = () => {
+    setVisibleCards(cardList.slice(0, MAX_ITEMS));
+
+    onButtonClick?.();
+  };
+
+  const hideContent = () => {
+    setVisibleCards(cardList.slice(0, initiallyShownCards));
+
+    onButtonClick?.();
+  };
+
+  const button = buttonText && (
+    <BpkButtonV2 onClick={onButtonClick}>{buttonText}</BpkButtonV2>
+  );
+
+  return (
+    <div className={getClassName('bpk-card-list')}>
+      <BpkSectionHeader
+        title={title}
+        description={description}
+        button={button}
+      />
+
+      <div className={getClassName('bpk-card-list--card-list')}>
+        <BpkBreakpoint query={BREAKPOINTS.MOBILE}>
+          {(isActive) => {
+            if (isActive) {
+              if (layoutMobile === 'rail') {
+                return <BpkCardListRail>{cardList}</BpkCardListRail>;
+              }
+              const { accessory } = props;
+
+              let modeProps = {};
+              if (accessory === BpkAccessoryTypes.Expand) {
+                modeProps = {
+                  accessory,
+                  expandText: props.expandText,
+                  onButtonClick: props.onButtonClick,
```

Comment:
Any reason these aren't extracted from the props object on L87?

Reply:
These props doesn't exist on the props until we narrow down the props object using `accessory === BpkAccessoryTypes...`